### PR TITLE
switched to successor for asciidoc maven plugin

### PR DESCRIPTION
**Pull request by grobmeier**
As described here:
https://docs.asciidoctor.org/maven-tools/latest/site-integration/converter-module-setup-and-configuration/

The asciidoctor-maven-plugin was replaced by the asciidoctor-converter-doxia-module which offers the same functionality. I think we can replace it, but we might have a change on the exclusion in log4j pom. 

So far I don't see any issues, but please let me know if there are some.

### CODE REVIEW

**Comment by vy** [comment]
@grobmeier, thanks so much for taking time to explore the AsciiDoc ecosystem. We should indeed switch to `asciidoctor-converter-doxia-module`, but not in `logging-parent`. Let me explain.

1. **`logging-parent` doesn't use `asciidoctor-maven-plugin` as a `maven-site-plugin` module**. Hence, the change is incorrect. `asciidoctor-maven-plugin` is still the recommended Maven plugin for standalone usages. Those who use `asciidoctor-maven-plugin` as a `maven-site-plugin` module, they should migrate to `asciidoctor-converter-doxia-module` instead. Since, it is only `logging-log4j2` that uses `maven-site-plugin`, you should submit two PRs there: one for `2.x` branch, one for `main` branch.
1. This is a breaking change. `logging-parent` itself and `logging-log4j2` still use `asciidoctor-maven-plugin` _managed by_ `logging-parent`. You cannot simply replace the old one, this will be a backward incompatible change causing warnings during build of every project using `logging-parent`, including `logging-parent` itself. Since they won't be able to resolve the correct version of `asciidoctor-maven-plugin`.
1. There are other usages of `asciidoctor-maven-plugin` in the `pom.xml` you touched. Please search around before replacing the first occurrence.

In short,
1. Could you revert this change please?
2. Could you submit two `logging-log4j2` PRs against `2.x` and `main` branches, please?
